include/seqan3/search/algorithm/search.hpp

```diff
@@ -28,14 +28,41 @@ namespace seqan3
 
 /*!\brief Search a query or a range of queries in an index.
  * \tparam index_t    Must model seqan3::FmIndex.
- * \tparam queries_t  Must model std::ranges::RandomAccessRange over the index's alphabet.
- *                    a range of queries must additionally model std::ranges::ForwardRange.
+ * \tparam queries_t  Must model std::ranges::RandomAccessRange over the index's alphabet and std::ranges::SizedRange.
+ *                    A range of queries must additionally model std::ranges::ForwardRange and std::ranges::SizedRange.
  * \param[in] queries A single query or a range of queries.
  * \param[in] index   String index to be searched.
  * \param[in] cfg     A configuration object specifying the search parameters (e.g. number of errors, error types,
  *                    output format, etc.).
- * \returns An object modelling std::ranges::Range containing the hits (the type depends on the specification
-            in `cfg`), or `void` if an on_hit delegate has been specified.
+ * \returns
+ * One object of the following type if one query was passed and a std::vector containing objects of the following type
+ * for each query if a range of queries was passed:
+ * <ul>
+ *     <li>If seqan3::search_cfg::index_cursor was configured:
+ *     <ul>
+ *         <li>A `std::vector<typename index_t::cursor_type>` containing index_cursors at the text positions were the
+ *             search was successful.</li>
+ *     </ul>
+ *     </li>
+ *     <li>If seqan3::search_cfg::text_position was configured:
+ *     <ul>
+ *         <li>For **single** text indices: A `std::vector<size_t>` representing text positions.</li>
+ *         <li>For indices over text **collections**: A `std::vector<std::vector<std::pair<size_t, size_t>>>` were the
+ *             first element specifies the text index in the collection and the second element contains the position
```

Comment:
why a vector of a vector? not just a vector? And the sentence after that is confusing. 
So is there a vector for each index with sequence id and position id?